docs/adr/0003-event-broker-for-subscription-platform.md

```diff
@@ -0,0 +1,110 @@
+    * B. Storing webhook and login activity in FxA Auth Service
+    * C. Storing webhook and login activity in FxA Event Broker
+
+## Decision Outcome
+
+Chosen Option: C. Implementing RP notification as a new FxA Event Broker service
+with webhook and login activity stored in FxA Event Broker, because
+
+* Less subscription platform timeline risk to store new data in new database vs.
+  modify existing FxA OAuth database.
+* Storing events at scale has database requirements that don't fit in well with
+  the limitations with MySQL encountered in FxA Auth/OAuth.
+* Having FxA Auth be the only store of which RP to notify would require each
+  notification to also include what RPs to notify, increasing the load on the
+  FxA Auth database.
+
+## Pros and Cons of the Options
+
+### A. Implementing RP notification as a FxA Auth Service feature w/webhook & login activity
+
+* Description
+  * FxA Auth Service has additional feature code added to perform webhook
+    deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+* Pros
+  * Auth service has direct access to subscription capability mapping and
+    OAuth webhook URLs.
+* Cons
+  * Event logs for each user are not well suited for existing MySQL database.
+  * Additional database schema migrations needed.
+  * Later features needing to query for user events could impact FxA Auth Service
+    availability.
+
+### B. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Auth Service
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Auth Service stores login activity and webhook mapping to OAuth Clients
+  * FxA Auth Server includes RPs that should be notified in the relevant
+    Service Notifications.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
+* Pros
+  * Division of event storing and distribution to a service that can be built
+    specifically for that function.
+* Cons
+  * Queries to FxA Auth Service required for each message to determine whether
+    to deliver to an RP could impact FxA Auth Service availability.
+  * FxA OAuth server should still have a record of which RPs a user has logged
+    into so a later issue to add this is still needed.
+
+### C. Implementing RP notification as a new FxA Event Broker service w/webhook & login activity in FxA Event Broker
+
+* Description
+  * FxA Event Broker performs webhook deliveries to RPs.
+  * FxA Event Broker caches new OAuth /clients request that indicates
+    subscription capabilities per RP.
+  * FxA Event Broker logs events and can determine which RPs a user has logged
+    into without any external service queries.
+* Pros
+  * Division of event storing and distribution to a service that can be built
+    specifically for that function.
+* Cons
+  * FxA OAuth server should still have a record of which RPs a user has logged
```

Comment:
The very last bullet point in the document is the one I wanted to ask! I imagine these records would not be accessible/able to be queried from, e.g., the auth-server or content-server (or other service) for account audit purposes.

Reply:
I think in the future the event broker will have an API so that the auth server can query for event log data on a user. But that's definitely beyond the scope of the subscription platform MVP.